### PR TITLE
initial docs and code

### PR DESCRIPTION
**Pull request by LeiGlobus**
TBD some info is missing

### CODE REVIEW

**Comment by LeiGlobus** [comment]
Some TODOs need to be fixed, but adding @joshbryan-globus to reviewer to start PR process

**Comment by LeiGlobus** [comment]
@joshbryan-globus FYI:

TL;DR Instead of testing it, take a look at some individual components to give general feedback, while I get the end-to-end working.

Some things to look at:

- Dockerfile/docker-compose.yaml
- Obviously missing how-tos in README.md or elsewhere
- The meat of the PR - provider/provider.py that contains the action logic
- local_db.py, a sqlite3 storage implementation, is long but can prob be skimmed as it's just a wrapper around a Key:Value store (action_id:[action_status, action_request])
- There are various TODOs scattered, some are to-be-removed useless code, others are in-progress fixes that  can be ignored (these should all go away before merge).  
- Currently the AP's CLIENT_ID and _SECRET are passed in via env variables (as detailed in README.md) it's the simplest way I can think of.  If you have a better alternative, let me know
- For Flows to be able to access a deployed SSH AP, it would need a https:// hostname.  I'm thinking Amazon ECS is an easy solution?  That's what I'm trying to set up.  If you think something else is better, let me know.

- The docker-compose/Dockerfile isn't running atm, so can not test it directly with a ready server (Current error is related to unknown client_id but it may be something else)

**Review by ghost — APPROVED** [review]
LGTM.  I know it's been a while, but since we haven't had an active need for this, I think we should merge this PR and revisit later if we need to add features.
